jpa/subsystem/src/main/java/org/jboss/as/jpa/beanmanager/PersistenceIntegrationWithCDI.java

```diff
@@ -0,0 +1,351 @@
+        beanConfigurator.addTransitiveTypeClosure(EntityManager.class);
+
+        try {
+            Class<? extends Annotation> scopeAnnotation = persistenceUnitMetadata.getClassLoader().loadClass(scope).asSubclass(Annotation.class);
+            beanConfigurator.scope(scopeAnnotation);
+
+            for (String qualifier : qualifiers) {
+                final Class<? extends Annotation> qualifierType = persistenceUnitMetadata.getClassLoader()
+                        .loadClass(qualifier)
+                        .asSubclass(Annotation.class);
+                beanConfigurator.addQualifier(ScopeProxy.createProxy(qualifierType));
+            }
+            Class<?> entityManagerClass = EntityManager.class;
+            beanConfigurator.beanClass(entityManagerClass);
+            beanConfigurator.produceWith(c -> {
+                        return new TransactionScopedEntityManager(
+                                persistenceUnitMetadata.getScopedPersistenceUnitName(),
+                                new HashMap<>(),
+                                integrationWithCDIBag.getEntityManagerFactory(),
+                                SynchronizationType.SYNCHRONIZED,
+                                integrationWithCDIBag.getTransactionSynchronizationRegistry(),
+                                integrationWithCDIBag.getTransactionManager());
+                    }
+            );
+        } catch (ClassNotFoundException e) {
+            throw JpaLogger.ROOT_LOGGER.classNotFound(e, persistenceUnitMetadata.getScopedPersistenceUnitName());
+        }
+    }
+
+    private void entityManagerFactory(
+            AfterBeanDiscovery afterBeanDiscovery,
+            PersistenceUnitMetadata persistenceUnitMetadata,
+            List<String> qualifiers,
+            IntegrationWithCDIBag integrationWithCDIBag) throws InstantiationException, IllegalAccessException {
+
+        String scope = applicationScoped;
+
+        // EntityManagerFactory setup
+        BeanConfigurator<EntityManagerFactory> beanConfigurator = afterBeanDiscovery.addBean();
+        beanConfigurator.addTransitiveTypeClosure(EntityManagerFactory.class);
+        // TODO: we cannot yet add the bean name for the EntityManagerFactory bean for each persistence unit when they
+        // * may be duplicated within different EE modules/submodules.  https://github.com/jakartaee/platform/issues/1135 is the
+        // * EE 12 tracker.
+        // Uncomment the following block and update when this TODO is addressed.
+        /**
+         String beanName = persistenceUnitMetadata.getScopedPersistenceUnitName().replace("#",".");
+         JpaLogger.ROOT_LOGGER.infof("Creating EntityManagerFactory CDI bean named %s for accessing persistence unit %s",
+         beanName, persistenceUnitMetadata.getPersistenceUnitName());
+         beanConfigurator.name(beanName);
+         **/
+        beanConfigurator.name(persistenceUnitMetadata.getPersistenceUnitName());
```

Comment:
Commenting out the call to **`beanConfigurator.name(persistenceUnitMetadata.getPersistenceUnitName())`** makes the test failures go away.  To be discussed in WildFly dev ml discussion.